### PR TITLE
Style rst directives: figure, quote, section headings

### PR DESCRIPTION
**Pull request by vsalvino**
* Add minimal styling for the following directives:
  * quotes
  * figures (images + caption) - this does not affect plain uncaptioned images.
* Remove section margin. This was causing a lot of empty space above page title.
* Add incremental spacing between sub-sections (h2, h3, etc.). This way h2s have more distance between them, h3s are a bit closer together, etc. to keep sub-sections and sub-sub-sections logically grouped together.
* Remove dead code (_component_rst.scss). That class was not being used anywhere.
* Added examples.

### CODE REVIEW

**Comment by vsalvino** [comment]
@tbrlpld could I get a quick review/approval to merge this in?

File: sass/_component_rst.scss [hunk 1]
```diff
@@ -1,305 +0,0 @@
-.rst-content {
```
Comment (Stormheg) [inline sass/_component_rst.scss:1]: I'm not sure how I feel about removing this. We don't currently use these styles, but they appear to be useful for styling the autodocs. See https://github.com/wagtail/sphinx_wagtail_theme/issues/131.

Let's just keep them for now, okay?

File: sass/_component_docutils.scss [hunk 2]
```diff
@@ -62,3 +53,33 @@ table.docutils {
     }
     margin-bottom: $spacer;
 }
+
+// Figures
+.figure {
```
Comment (Stormheg) [inline sass/_component_docutils.scss:58]: Perhaps we should consider extracting these styles into their own `.scss` files?

That would provide more long term structure I think.

**Comment by Stormheg** [comment]
Rebased on latest main +
- Pulled the new styles into their own files
- Altered examples so they render properly with the new MyST parser.

Going to merge this now. Thanks @vsalvino! ❤️

**Pull request merged**